vibe_code_apps_22/scripts/a04v2_phase0_freeze.py

```
@@ -0,0 +1,91 @@
+    out_dir.mkdir(parents=True, exist_ok=True)
+    # Immutable copy reference (do not mutate source)
+    copy_path = out_dir / f"IMMUTABLE_{A04_NAME}"
+    if not copy_path.is_file() or sha256_file(copy_path) != idf_sha:
+        copy_path.write_bytes(idf.read_bytes())
+
+    git_sha = Path(_APP).resolve().parents[1]
+    # worktree root is parents[1] from vibe_code_apps_22
+    import subprocess
+
+    repo = _APP.parent
+    g = subprocess.run(["git", "rev-parse", "HEAD"], cwd=repo, capture_output=True, text=True)
```

Ruff (0.16.1)

[error] Starting a process with a partial executable path

(S607)